src/components/MovieDetails/MovieDetails.tsx

```diff
@@ -0,0 +1,50 @@
+import { useEffect, useRef, useState } from "react";
+import { finedFilm } from "../../helpers";
+import { Film } from "../../interfaces";
+import "./style.scss";
+
+interface MovieDetailsProps {
+  closeFilm(): void;
+  filmID: string;
+}
+
+const MovieDetails: React.FC<MovieDetailsProps> = ({ closeFilm, filmID }) => {
+  const [movie, setMovie] = useState<Film>(finedFilm(filmID));
```

Comment:
just put a whole film as a prop for MovieDetails, and you do not need this state,extra search/find, useEffect, and useRef here at all.